File: apps/web/src/shared/entities/bet.entity.ts
export interface PredictionEntity {
  matchId: number;
  homeScore: number;
  awayScore: number;
}

export interface BetEntity {
  id: number;
  roundId: number;
  ticketCode: string | null;
  name: string;
  phone: string;
  predictions: Record<number, PredictionEntity>;
  points: number;
  createdAt: string;
}

export interface CreateBetResponse {
  bet: BetEntity;
  payment: {
    pixCopyPaste: string;
    pixQrCodeBase64: string;
    amount: number;
    expiration: string;
  };
}

File: apps/api/src/main/java/com/bolao/bet/entities/BetEntity.java
package com.bolao.bet.entities;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "bets")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class BetEntity {

  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private Long id;

  @Column(name = "round_id")
  private Long roundId;

  private String name;

  private String phone;

  @Column(name = "ticket_code", unique = true)
  private String ticketCode;

  private Integer points;

  @Column(name = "created_at")
  private LocalDateTime createdAt;

  @Enumerated(EnumType.STRING)
  private Bet.PaymentStatus status;

  @OneToMany(mappedBy = "bet", cascade = CascadeType.ALL, fetch = FetchType.EAGER)
  @Builder.Default
  private List<PredictionEntity> predictions = new ArrayList<>();
}
